lib/ddtrace/contrib/action_cable/event.rb

```diff
@@ -0,0 +1,64 @@
+require 'ddtrace/contrib/analytics'
+require 'ddtrace/contrib/active_support/notifications/event'
+require 'ddtrace/contrib/action_cable/ext'
+
+module Datadog
+  module Contrib
+    module ActionCable
+      # Defines basic behaviors for an ActiveRecord event.
+      module Event
+        def self.included(base)
+          base.send(:include, ActiveSupport::Notifications::Event)
+          base.send(:extend, ClassMethods)
+        end
+
+        # Class methods for ActionCable events.
+        # Note, they share the same process method and before_trace method.
+        module ClassMethods
+          def subscription(*args)
+            super.tap do |subscription|
+              subscription.before_trace { ensure_clean_context! }
```

Comment:
Given ActionCable does some async stuff, is it possible that a thread executes this for channel A, then B, then A again? I'm just trying to think if there are any scenarios where we might drop trace contexts erroneously given the async and thread reuse behavior.